contracts/src/challengeV2/libraries/EdgeChallengeManagerLib.sol

```diff
@@ -404,24 +417,58 @@ library EdgeChallengeManagerLib {
     /// @param oneStepProofEntry    The one step proof contract that defines how machine states are hashed
     /// @param expectedEndHeight    The expected end height of an edge. Layer zero block edges have predefined heights.
     /// @param numBigStepLevel      The number of big step levels in this challenge
+    /// @param initialStakeAmount   The initial stake amount for layer zero edges with no rivals
+    /// @param stakeAmountSlope     The increase in stake amount caused by each additional rival edge
     function createLayerZeroEdge(
         EdgeStore storage store,
         CreateEdgeArgs calldata args,
         AssertionReferenceData memory ard,
         IOneStepProofEntry oneStepProofEntry,
         uint256 expectedEndHeight,
-        uint8 numBigStepLevel
+        uint8 numBigStepLevel,
+        uint256 initialStakeAmount,
+        uint256 stakeAmountSlope
     ) internal returns (EdgeAddedData memory) {
         // each edge type requires some specific checks
         (ProofData memory proofData, bytes32 originId) =
             layerZeroTypeSpecificChecks(store, args, ard, oneStepProofEntry, numBigStepLevel);
         // all edge types share some common checks
         (bytes32 startHistoryRoot) = layerZeroCommonChecks(proofData, args, expectedEndHeight);
         // we only wrap the struct creation in a function as doing so with exceeds the stack limit
+        // we also don't pass in stakeAmount to avoid the stack limit
         ChallengeEdge memory ce = toLayerZeroEdge(originId, startHistoryRoot, args);
+
+        // calculate, check and set stake amount
+        ce.stakeAmount = checkStakeAmount(store, ce, args, initialStakeAmount, stakeAmountSlope);
+
         return add(store, ce);
     }
 
+    /// @notice Calculate stake amount for an edge and require it is less than the max stake amount
+    function checkStakeAmount(
+        EdgeStore storage store,
+        ChallengeEdge memory ce,
```

Comment:
seems like we can pass only the mutual id instead of the whole edge struct

Reply:
51b9a17c9512c1362acef69e21104ebcde68107f